lib/Macros/Sources/ObservationMacros/ObservableMacro.swift

```diff
@@ -281,13 +280,29 @@ extension ObservableMacro: ExtensionMacro {
       return []
     }
 
-    let decl: DeclSyntax = """
-      extension \(raw: type.trimmedDescription): \(raw: qualifiedConformanceName) {}
-      """
+    if let availability = declaration.availability {
+      let emptyBlock: MemberDeclBlockSyntax = "{ }"
 
-    return [
-      decl.cast(ExtensionDeclSyntax.self)
-    ]
+      return [
+        ExtensionDeclSyntax(
+          leadingTrivia: .newline,
+          attributes: availability,
+          extensionKeyword: .keyword(.extension),
+          extendedType: type,
+          inheritanceClause: TypeInheritanceClauseSyntax(inheritedTypeCollection: InheritedTypeListSyntax([InheritedTypeSyntax(typeName: observableConformanceType)])),
+          memberBlock: emptyBlock,
+          trailingTrivia: .newline
+        )
+      ]
+    } else {
+      let decl: DeclSyntax = """
+        extension \(raw: type.trimmedDescription): \(raw: qualifiedConformanceName) {}
```

Comment:
It's unfortunate to have to build the `ExtensionDeclSyntax` directly in one case, and use interpolation in the other. Is there no way to use this `else` path all the time, interpolating in `availability`?

Reply:
perhaps, I can give that a whirl.

Reply:
ah yes that fails the `decl.cast` because it has availability attached.

Reply:
this seems to work via the `with` syntax to alter the availability.